### PR TITLE
Initialize package and tooling

### PR DESCRIPTION
**Pull request by omgovich**
### 👋

1. Copied data from the primary repo (no changes).

2. Converted the source to TS and defined the types. I'm sure we need that to avoid errors (like typos in object keys) that contributors might make accidentally. Also, it should help other developers to use our package in their TS-projects.

3. Added ES Lint and Prettier to catch and fix code style. Should help us to keep the file consistent. 

4. Configured package.json fields (files, main, keywords, etc)

I also thought to add simple tests, but TS type-check already does all of the work.

### Next steps:

Setup CI (GitHub Action) to test linting and TS automatically. Should make our PR review process faster.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,16 @@
 # similar-npm-packages
+
 Similar packages suggestions for the NPM ecosystem
+
+## Install
+
+```
+npm install similar-npm-packages --save
```
Comment (pastelsky) [inline README.md:8]: Do you want me to reserve this name?

**Review by pastelsky — APPROVED** [review]
Looks good. Love that we're doing typescript with this.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  "description": "Similar packages suggestions for the NPM ecosystem",
+  "main": "dist/index.js",
+  "scripts": {
+    "lint": "eslint src",
+    "check-types": "tsc --noEmit true",
+    "test": "npm run lint && npm run check-types",
+    "build": "rm -rf ./dist/* && tsc --outDir dist",
+    "prepublishOnly": "npm run build"
+  },
+  "repository": "pastelsky/similar-npm-packages",
```
Comment (pastelsky) [inline package.json:13]: Could we add the full github link here?

File: README.md (hunk 1 above)
Comment (omgovich) [reply README.md:8]: I do)

File: package.json (hunk 2 above)
Comment (omgovich) [reply package.json:13]: As far I know all systems (including NPM and bundlephobia) understand the short statements.
But I could insert the full link if you want.

File: README.md (hunk 1 above)
Comment (omgovich) [reply README.md:8]: BTW: Are we going to have NodeJS only packages here? Maybe `similar-npm-packages` isn't the best name if we're going to work with browsers packages only.

File: package.json (hunk 2 above)
Comment (pastelsky) [reply package.json:13]: Ah, I forget how much of a default github has become now. That's fine then.

File: README.md (hunk 1 above)
Comment (pastelsky) [reply README.md:8]: We have some node.js packages in recommendations as well. Folks in the electron community find it useful to see sizes for these. Let's be stack agnostic for now.

Comment (omgovich) [reply README.md:8]: Alight. We should "book" the name then)

Comment (pastelsky) [reply README.md:8]: `similar-npm-packages` is ours 🥳

**Comment by pastelsky** [comment]
LGTM for merge.

File: README.md (hunk 1 above)
Comment (omgovich) [reply README.md:8]: Nice!

**Pull request merged**